src/bot/exts/rdap.py

```diff
@@ -0,0 +1,219 @@
+
+        try:
+            async with self.bot.http_session.get(url) as response:
+                if response.status == 404:
+                    await ctx.send(f"❌ No results found for `{query}`.")
+                    return
+                if response.status != 200:
+                    log.warning(f"RDAP lookup failed for {query}: HTTP {response.status}")
+                    await ctx.send(f"❌ Error fetching RDAP data: HTTP {response.status}")
+                    return
+
+                data = await response.json()
+        except Exception as e:
+            log.exception(f"Error performing RDAP lookup for {query}: {e}")
+            await ctx.send("❌ An error occurred while fetching RDAP data.")
+            return
+
+        # Handle "Thin" registries (e.g., .com, .net) which provide a "related" link to the full RDAP info
+        if query_type == "domain":
+            for link in data.get("links", []):
+                if link.get("rel") == "related" and link.get("type") == "application/rdap+json":
+                    related_url = link.get("href")
+                    if related_url:
+                        log.debug(f"Following related RDAP link: {related_url}")
+                        try:
+                            async with self.bot.http_session.get(related_url) as related_response:
+                                if related_response.status == 200:
+                                    data = await related_response.json()
+                        except Exception:
+                            log.warning(f"Failed to follow related RDAP link: {related_url}", exc_info=True)
+                        break
+
+        result_data: dict[str, Any] = {}
+        title = f"RDAP Lookup: {query}"
+
+        if query_type == "domain":
+            model = RDAPDomain(**data)
+            registrar = model.get_entity_by_role("registrar")
+            registrant = model.get_entity_by_role("registrant")
+            abuse = model.get_entity_by_role("abuse")  # Sometimes abuse contact is separate
+
+            iana_id = None
+            if registrar and registrar.publicIds:
+                for pid in registrar.publicIds:
+                    if "IANA" in pid.get("type", ""):
+                        iana_id = pid.get("identifier")
+                        break
+
+            result_data = {
+                "Domain Name": model.ldhName,
+                "Registrar": registrar.contact_info.get("name") if registrar else None,
+                "IANA ID": iana_id,
+                "Registered": model.registration_date,
+                "Abuse Email": abuse.contact_info.get("email") if abuse else None,
+            }
+
+            if model.nameservers:
+                ns_list = [str(ns.get("ldhName")) for ns in model.nameservers if ns.get("ldhName")]
+                result_data["Nameservers"] = ", ".join(ns_list[:3]) + ("..." if len(ns_list) > 3 else "")
+
+        elif query_type == "ip":
+            model = RDAPIP(**data)
```

Comment:
```suggestion
            model = RDAPIP.model_validate(data)
```

Reply:
Hm, I wasn't aware they did the same thing. If that's the case, then I suppose we can leave it up to personal preference - though `model_validate` seems more explicit to me.